services/workflows-service/src/document/document.service.ts

```diff
@@ -269,13 +269,9 @@ export class DocumentService {
     });
   }
 
-  async getDocumentTrackerByWorkflowId(
-    projectId: TProjectId,
-    workflowDefinitionId: string,
-    workflowRuntimeDataId: string,
-  ) {
-    const uiDefinition = await this.uiDefinitionService.getByWorkflowDefinitionId(
-      workflowDefinitionId,
+  async getDocumentTrackerByWorkflowId(projectId: TProjectId, workflowId: string) {
+    const uiDefinition = await this.uiDefinitionService.getByRuntimeId(
+      workflowId,
       'collection_flow',
       [projectId],
     );
```

Comment:
Not a blocker but couldn't we get both the workflow and the ui definition in a single query at the end?